build/source/driver/multi_driver.f90

```diff
@@ -128,70 +143,78 @@ program multi_driver
+type(gru_hru_double)             :: mparStruct                 ! x%gru(:)%hru(:)%var(:)     -- model parameters
+! define the primary data structures (variable length vectors)
+type(gru_hru_intVec)             :: indxStruct                 ! x%gru(:)%hru(:)%var(:)%dat -- model indices
+type(gru_hru_doubleVec)          :: progStruct                 ! x%gru(:)%hru(:)%var(:)%dat -- model prognostic (state) variables
+type(gru_hru_doubleVec)          :: diagStruct                 ! x%gru(:)%hru(:)%var(:)%dat -- model diagnostic variables
+type(gru_hru_doubleVec)          :: fluxStruct                 ! x%gru(:)%hru(:)%var(:)%dat -- model fluxes
+! define the basin-average structures
+type(gru_double)                 :: bparStruct                 ! x%gru(:)%var(:)            -- basin-average parameters
+type(gru_doubleVec)              :: bvarStruct                 ! x%gru(:)%var(:)%dat        -- basin-average variables
 ! define the ancillary data structures
-type(spatial_double)      :: dparStruct    ! x%hru(:)%var(:)     -- default model parameters
-! define counters
-integer(i4b)              :: iVar                           ! index of a model variable 
-integer(i4b)              :: iHRU,jHRU,kHRU                 ! index of the hydrologic response unit
-integer(i4b)              :: nHRU                           ! number of hydrologic response units
-integer(i4b)              :: iStep=0                        ! index of model time step
-integer(i4b)              :: jStep=0                        ! index of model output
+type(gru_hru_double)             :: dparStruct                 ! x%gru(:)%hru(:)%var(:)     -- default model parameters
+! define indices
+integer(i4b)                     :: iVar                       ! index of a model variable 
+integer(i4b)                     :: iStruct                    ! loop through data structures
+integer(i4b)                     :: iGRU
+integer(i4b)                     :: iHRU,jHRU,kHRU             ! index of the hydrologic response unit
+integer(i4b)                     :: nGRU                       ! number of grouped response units
+integer(i4b)                     :: nHRU                       ! number of global hydrologic response units
+integer(i4b)                     :: hruCount                   ! number of local hydrologic response units
+integer(i4b)                     :: iStep=0                    ! index of model time step
+integer(i4b)                     :: jStep=0                    ! index of model output
+integer(i4b)                     :: ix_gru                     ! index of GRU that corresponds to the global HRU
+integer(i4b)                     :: ix_hru                     ! index of local HRU that corresponds to the global HRU
+integer(i4b)                     :: jx_gru                     ! mapping from a previously used forcing file: index of GRU that corresponds to the global HRU
```

Comment:
Can we use a better index than `jx_gru` since this appears to be used for a very specific case rather than a loop counter (which is what I would associate most i, j, k variables with)?